### ISSUE TITLE
Worse performance of tensorGraph implementation of Weave

### ISSUE THREAD

**Issue by miaecle** [deepchem/deepchem#659]
`dc.models.WeaveTensorGraph` is a TensorGraph based implementation of the original Weave model(`deepchem.nn.AlternateSequentialWeaveGraph`), using exactly the same layer structure. But the performance on delaney(ESOL) does not match: valid r2 0.82(WeaveTensorGraph, 100 epochs) vs 0.89(SequentialWeaveGraph, 50 epochs), 50-epoch performance is even worse for WeaveTensorGraph, all other hyper parameters are set to be the same.
 
Is there any subtle differences(like optimizer, loss) between tensorgraph and the old version model platform(sequentialgraph) that can explain the performance gap?
cc @lilleswing

**Comment by lilleswing** [comment]
@miaecle I'll look through it today.  My initial guess is that despite using the same "layers" we are using different initialization constants.

**Comment by miaecle** [comment]
@lilleswing That will be super helpful, thanks a lot!

**Comment by patrickhop** [comment]
Have you looked into the learning rate decay schedule? I can't speak to Weaves, but that was one of the differences on the GraphConv side.

**Comment by vid33** [comment]
Hi all. I'm new here and have been looking the DeepChem code - appreciating all the amazing work that's gone into the project -  and have just come across this issue. Have you been able to figure out what the reason is? 

In order to understand better what's going on, I've also tried to unsucessfully figure out a simple way to read out the outputs of intermediate layers. Does the TensorGraph class provide an easy way to do this?

Thanks a lot!

**Comment by lilleswing** [comment]
Hi @vid33, TensorGraph does not currently have an easy way of viewing the intermediate layers.  What I would recommend though is setting TensorGraph.outputs to a singleton list of the layer you are interested in viewing.

When you do this TensorGraph.predict_proba will return the evaluation for the layer you are interested in.  Here is a working example of it in action.

```python
"""
Script that trains graph-conv models on Tox21 dataset.
"""
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

import numpy as np

import tensorflow as tf

tf.set_random_seed(123)
import deepchem as dc
from tox21_datasets import load_tox21
from deepchem.models.tensorgraph.models.graph_models import GraphConvTensorGraph

model_dir = "/tmp/graph_conv"

# Load Tox21 dataset
tox21_tasks, tox21_datasets, transformers = load_tox21(featurizer='GraphConv')
train_dataset, valid_dataset, test_dataset = tox21_datasets

# Batch size of models
batch_size = 50

model = GraphConvTensorGraph(
    len(tox21_tasks), batch_size=batch_size, mode='classification')

model.fit(train_dataset, nb_epoch=10)
key_layer = model.layers['Dense_2']
model.outputs = [key_layer]
X = model.predict_proba(train_dataset)
print(X)
```

Eventually we want to be able to pass a list of layers into the TensorGraph#predict* functions as an optional parameter but haven't gotten around to doing the work yet.

**Comment by vid33** [comment]
Thanks, that probably saved me hours! I'll let you know if I manage to understand where the above discrepancy is coming from.

**Comment by miaecle** [comment]
Found a typo in `WeaveTensorGraph` which turns off the `gaussian_expand` function, that cause the worse performance. Thanks @patrickhop and @lilleswing for providing the help! 
@vid33 Welcome! I basically agree with @lilleswing , this seems to be the easy way to look at intermediate output.
I will go ahead and close this issue.
